### DISCUSSION TITLE
[Script request] Odoo

### DISCUSSION THREAD

**Discussion by morno** [community-scripts/ProxmoxVE#512] [Request script]
### Application Name

Odoo 18.0

### Website

https://www.odoo.com/page/download

### Description

Hello Community,

I would like to request a script to install Odoo v18.0 on Proxmox VE. The script should aim to simplify the installation process while maintaining flexibility for customization.

   1. Environment Configuration:
        Support for LXC (Debian/Ubuntu templates) or KVM environments.
        Set up a clean and secure environment optimized for Odoo.

   2 .Dependencies:
        Automatically install all required dependencies (e.g., Python, PostgreSQL).
        Proper versioning for compatibility with Odoo v18.0.

  3. Database Setup:
        Install and configure PostgreSQL, creating a database user and database for Odoo.

   4. Odoo Installation:
        Fetch and install the latest stable release of Odoo v18.0.
        Configure a dedicated service for Odoo (e.g., systemd or init.d).

  5. Reverse Proxy and SSL (Optional):
        Option to set up Nginx as a reverse proxy.
        Automatically configure and obtain an SSL certificate using Let's Encrypt.

   6. Parameters for Customization:
        Allow specifying installation paths, database settings, and domain names.
        Options for multi-company and multi-database setups.

  7. User-Friendly:
        Clear and detailed log outputs.
        Automated error handling and rollback mechanism for failed setups.

   8. Documentation:
        Basic instructions on running the script and verifying the installation.

This script will be incredibly useful for Proxmox VE users looking to streamline the deployment of Odoo v18.0 for their businesses or projects.

Thank you for considering this request! Let me know if additional details or clarifications are needed.

Here are my notes to get it working in Ubuntu 24.04 VM

[Odoo_v18_Installation_Guide.txt](https://github.com/user-attachments/files/17907382/Odoo_v18_Installation_Guide.txt)

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by MickLesk** [comment]
And where can i get it from?
On the page you have to fill in Your Company, Your Name and so on (also in the Community Script).

I remember the request from the tteck repo, and I still think it's very much geared towards “revenue”. It doesn't look like a community thing, it just looks like profit.

**Comment by jsapede** [comment]
installed it through docker : 

https://github.com/odoo/docker/tree/master/18.0

maybe dockerfile would be a good start. however, community edition lacks functionnalities

**Comment by Sedation6612** [comment]
+1!

**Comment by MickLesk** [comment]
Ready for Test:
https://github.com/community-scripts/ProxmoxVED/issues/356

Please checkout the link and put feedback into this one.

**Comment by MickLesk** [comment]
Merged
